nevergrad/functions/images/core.py

```diff
@@ -50,16 +53,46 @@ def __init__(
         image = PIL.Image.open(path).resize((self.domain_shape[0], self.domain_shape[1]), PIL.Image.ANTIALIAS)
         self.data = np.asarray(image)[:, :, :3]  # 4th Channel is pointless here, only 255.
         # parametrization
-        array = ng.p.Array(init=128 * np.ones(self.domain_shape), mutable_sigma=True)
-        array.set_mutation(sigma=35)
-        array.set_bounds(lower=0, upper=255.99, method="clipping", full_range_sampling=True)
-        max_size = ng.p.Scalar(lower=1, upper=200).set_integer_casting()
-        array.set_recombination(ng.p.mutation.Crossover(axis=(0, 1), max_size=max_size)).set_name("")  # type: ignore
-        super().__init__(loss(reference=self.data), array)
+        if not with_pgan:
+            array = ng.p.Array(init=128 * np.ones(self.domain_shape), mutable_sigma=True)
+            array.set_mutation(sigma=35)
+            array.set_bounds(lower=0, upper=255.99, method="clipping", full_range_sampling=True)
+            max_size = ng.p.Scalar(lower=1, upper=200).set_integer_casting()
+            array.set_recombination(ng.p.mutation.Crossover(axis=(0, 1), max_size=max_size)).set_name("")  # type: ignore
+            super().__init__(loss(reference=self.data), array)
+        else:
+            self.pgan_model = torch.hub.load(
+                "facebookresearch/pytorch_GAN_zoo:hub",
+                "PGAN",
+                model_name="celebAHQ-512",
+                pretrained=True,
+                useGPU=False,
+            )
+            self.domain_shape = (1, 512)  # type: ignore
+            initial_noise = np.random.normal(size=self.domain_shape)
+            array = ng.p.Array(init=initial_noise, mutable_sigma=True)
+            array.set_mutation(sigma=35.0)
+            array.set_recombination(ng.p.mutation.Crossover(axis=(0, 1))).set_name("")
+            self._descriptors.pop("use_gpu", None)
+            super().__init__(self._loss, array)
```

Comment:
shouldn't the with_pgan part be in a different class? Here self.pgan_model will not always be defined so _generate_images will fail if you created the instance with pgan_model = False.

Reply:
Why would we call _generate_images if there is no pgan_model ?

Reply:
After offline discussion, I renamed functions.